packages/sdks/web-component/src/lib/descope-wc/DescopeWc.ts

```diff
@@ -157,31 +162,72 @@ class DescopeWc extends BaseDescopeWc {
       }
     | undefined;
 
-  async loadSdkScripts() {
-    const flowConfig = await this.getFlowConfig();
-    const scripts = flowConfig.sdkScripts;
-
-    scripts?.forEach(async (script) => {
-      const module = await loadSdkScript(script.id);
-      module(
-        script.initArgs as any,
-        { baseUrl: this.baseUrl },
-        (result: string) => {
-          // update the context with the result, under the `resultKey` key
-          this.dispatchEvent(
-            new CustomEvent('components-context', {
-              detail: {
-                // we store the result with script.id prefix to avoid conflicts with other scripts results
-                // that may have the same key
-                [getScriptResultPath(script.id, script.resultKey)]: result,
-              },
-              bubbles: true,
-              composed: true,
-            }),
-          );
+  #scriptsRegistry = new Map<string, ScriptModule>();
+
+  loadSdkScripts(scripts: ClientScript[]) {
+    if (!scripts?.length) {
+      return null;
+    }
+
+    const createScriptCallback =
+      (
+        script: {
+          id: string;
+          resultKey?: string;
         },
-      );
+        resolve: (value: any) => void,
+      ) =>
+      (result: string) => {
+        this.dispatchEvent(
+          // update the context with the result, under the `resultKey` key
+          new CustomEvent('components-context', {
+            detail: {
+              // we store the result with script.id prefix to avoid conflicts with other scripts results
+              // that may have the same key
+              [getScriptResultPath(script.id, script.resultKey)]: result,
+            },
+            bubbles: true,
+            composed: true,
+          }),
+        );
+        resolve(script.id);
+      };
+
+    const promises = Promise.all(
+      scripts?.map(async (script) => {
+        if (this.#scriptsRegistry.has(script.id)) {
+          this.loggerWrapper.debug('Script already loaded', script.id);
+          const m = this.#scriptsRegistry.get(script.id);
+          m.start?.();
+          return m;
+        }
+        const module = await loadSdkScript(script.id);
+        return new Promise((resolve, reject) => {
+          try {
+            const moduleRes = module(
+              script.initArgs as any,
+              { baseUrl: this.baseUrl },
+              createScriptCallback(script, resolve),
+            );
+            if (moduleRes) {
+              this.#scriptsRegistry.set(script.id, moduleRes);
+              this.nextRequestStatus.subscribe(() => {
+                this.loggerWrapper.debug('Unloading script', script.id);
+                moduleRes.stop?.();
+              });
+            }
+          } catch (e) {
+            reject(e);
+          }
+        });
+      }),
+    );
+
+    const toPromise = new Promise((resolve) => {
+      setTimeout(resolve, SDK_SCRIPTS_LOAD_TIMEOUT);
```

Comment:
it is possible we may want to `reject` the promis here if the script failed or timeouted to load

Reply:
This is why I have the race